src/Toolkit/Toolkit/UI/Controls/PopupViewer/PopupViewer.Windows.cs

```diff
@@ -56,6 +56,32 @@ private static void OnPopupManagerPropertyChanged(DependencyObject d, Dependency
             ((PopupViewer)d).Popup = ((PopupViewer)d).PopupManager?.Popup;
 #pragma warning restore CS0618 // Type or member is obsolete
         }
+
+        internal static PopupViewer? GetPopupViewerParent(DependencyObject? child) => GetParent<PopupViewer>(child);
+
+        private static T? GetParent<T>(DependencyObject? child) where T : DependencyObject
+        {
+            // Traverse up the visual tree to find the PopupViewer parent
+            while (child is not null)
+            {
+                if (child is FrameworkContentElement contentElement)
+                {
+                    // Use the logical parent for FrameworkContentElement instances
+                    child = contentElement.Parent ?? VisualTreeHelper.GetParent(contentElement);
+                }
+                else
```

Comment:
Should this then check that `child` is a `Visual` ?

Reply:
That would be clearer I think.